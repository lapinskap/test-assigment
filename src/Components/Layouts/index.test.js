import React from 'react';
import renderer from 'react-test-renderer';
// import '../../tests/setupTests';
import {
  getLayout, LAYOUT_THREE_COLUMNS, LAYOUT_TWO_COLUMNS, LAYOUT_ONE_COLUMN,
} from './index';

describe('<Tiles />', () => {
  const getChildrenNumber = (numberOfElements) => [...Array(numberOfElements).keys()].map((el) => (
    <div key={el}>
      test
      {el}
    </div>
  ));

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('returns nothing on wrong layout name', () => {
    global.console = { error: jest.fn() };
    const result = getLayout('wrong_layout_name', getChildrenNumber(10));
    expect(result)
      .toBe(null);
    expect(console.error)
      .toBeCalledTimes(1);
  });

  it('renders correctly 2 columns layout', () => {
    expect(renderer
      .create(getLayout(LAYOUT_TWO_COLUMNS, getChildrenNumber(10), { sm: 4, md: 4, lg: 4 }))
      .toJSON())
      .toMatchSnapshot();
    expect(renderer
      .create(getLayout(LAYOUT_TWO_COLUMNS, getChildrenNumber(10), { sm: [3, 7], md: [5, 5], lg: [4, 2] }))
      .toJSON())
      .toMatchSnapshot();
  });

  it('renders correctly 1 columns layout with border', () => {
    expect(renderer
      .create(getLayout(LAYOUT_ONE_COLUMN, getChildrenNumber(10), {}, 'key_1', true))
      .toJSON())
      .toMatchSnapshot();
    expect(renderer
      .create(getLayout(LAYOUT_ONE_COLUMN, getChildrenNumber(10), { sm: [12], md: [12], lg: [12] }, 'key_1', true))
      .toJSON())
      .toMatchSnapshot();
  });

  it('renders correctly 3 columns layout', () => {
    expect(renderer
      .create(getLayout(LAYOUT_THREE_COLUMNS, getChildrenNumber(10), { sm: [4, 4, 4], md: [4, 4, 4], lg: [3, 3, 4] }, 'key_1'))
      .toJSON())
      .toMatchSnapshot();
    expect(renderer
      .create(getLayout(LAYOUT_THREE_COLUMNS, getChildrenNumber(10), { sm: 4, md: 3, lg: 4 }, 'key_1'))
      .toJSON())
      .toMatchSnapshot();
  });

  it('renders null when no children', () => {
    const Layout = getLayout(LAYOUT_THREE_COLUMNS, [], {}, 'key_1');
    const tree = renderer
      .create(Layout)
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});
