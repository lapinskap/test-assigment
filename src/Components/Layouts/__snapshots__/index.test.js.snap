// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tiles /> renders correctly 1 columns layout with border 1`] = `
<div
  className="border p-2 mt-1 mb-1"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        0
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        1
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        2
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        3
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        4
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        5
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        6
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        7
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        8
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div>
        test
        9
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tiles /> renders correctly 1 columns layout with border 2`] = `
<div
  className="border p-2 mt-1 mb-1"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        0
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        1
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        2
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        3
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        4
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        5
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        6
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        7
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        8
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-12 col-lg-12"
    >
      <div>
        test
        9
      </div>
    </div>
  </div>
</div>
`;

exports[`<Tiles /> renders correctly 2 columns layout 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        0
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        1
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        2
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        3
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        4
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        5
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        6
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        7
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        8
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        9
      </div>
    </div>
  </div>,
]
`;

exports[`<Tiles /> renders correctly 2 columns layout 2`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm-3 col-md-5 col-lg-4"
    >
      <div>
        test
        0
      </div>
    </div>
    <div
      className="col-sm-7 col-md-5 col-lg-2"
    >
      <div>
        test
        1
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-3 col-md-5 col-lg-4"
    >
      <div>
        test
        2
      </div>
    </div>
    <div
      className="col-sm-7 col-md-5 col-lg-2"
    >
      <div>
        test
        3
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-3 col-md-5 col-lg-4"
    >
      <div>
        test
        4
      </div>
    </div>
    <div
      className="col-sm-7 col-md-5 col-lg-2"
    >
      <div>
        test
        5
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-3 col-md-5 col-lg-4"
    >
      <div>
        test
        6
      </div>
    </div>
    <div
      className="col-sm-7 col-md-5 col-lg-2"
    >
      <div>
        test
        7
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-3 col-md-5 col-lg-4"
    >
      <div>
        test
        8
      </div>
    </div>
    <div
      className="col-sm-7 col-md-5 col-lg-2"
    >
      <div>
        test
        9
      </div>
    </div>
  </div>,
]
`;

exports[`<Tiles /> renders correctly 3 columns layout 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-3"
    >
      <div>
        test
        0
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-3"
    >
      <div>
        test
        1
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        2
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-3"
    >
      <div>
        test
        3
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-3"
    >
      <div>
        test
        4
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        5
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-3"
    >
      <div>
        test
        6
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-3"
    >
      <div>
        test
        7
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    >
      <div>
        test
        8
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-4 col-lg-3"
    >
      <div>
        test
        9
      </div>
    </div>
    <div
      className="col-sm-4 col-md-4 col-lg-3"
    />
    <div
      className="col-sm-4 col-md-4 col-lg-4"
    />
  </div>,
]
`;

exports[`<Tiles /> renders correctly 3 columns layout 2`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        0
      </div>
    </div>
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        1
      </div>
    </div>
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        2
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        3
      </div>
    </div>
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        4
      </div>
    </div>
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        5
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        6
      </div>
    </div>
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        7
      </div>
    </div>
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        8
      </div>
    </div>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    >
      <div>
        test
        9
      </div>
    </div>
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    />
    <div
      className="col-sm-4 col-md-3 col-lg-4"
    />
  </div>,
]
`;

exports[`<Tiles /> renders null when no children 1`] = `null`;
